app/src/main/java/ch/epfl/sweng/rps/ui/game/GameFragment.kt

```diff
@@ -33,40 +31,54 @@ class GameFragment : Fragment() {
         binding.rockRB.setOnClickListener { rpsPressed(Hand.ROCK) }
         binding.paperRB.setOnClickListener { rpsPressed(Hand.PAPER) }
         binding.scissorsRB.setOnClickListener { rpsPressed(Hand.SCISSORS) }
-
+        matchViewModel.cumulativeScore.observe(
+            viewLifecycleOwner
+        ) {
+            binding.opponentMatchData.currentPoints.text =
+                matchViewModel.computerPlayerCurrentPoints
+            binding.userMatchData.currentPoints.text = matchViewModel.userPlayerCurrentPoints
+        }
     }
 
     private fun rpsPressed(hand: Hand) {
-        matchViewModel.playHand(hand,
-            updateUIResultCallback = {
-                selectComputerChoice(
-                    matchViewModel.gameService?.currentRound?.hands?.get(
-                        matchViewModel.computerPlayer!!.computerPlayerId
-                    )!!
-                )
-            },
-            resultNavigationCallback = {
-                findNavController().navigate(R.id.action_gameFragment_to_gameResultFragment)
-            },
-            isGameOverCallback = {
-                gameOverNavigation()
-            }
-        )
+        // the button activates an asynchronous tasks
+        // while the task is still running the clicking on the other radio buttons should NOT
+        // start a new procedure and that's why the if statement in needed below
+        if (matchViewModel.job == null ||
+            (matchViewModel.job != null && !matchViewModel.job?.isActive!!)
+        ) {
+            matchViewModel.managePlayHand(hand,
+                opponentsMoveUIUpdateCallback = {
+                    opponentMoveUIUpdate(
+                        matchViewModel.gameService?.currentRound?.hands?.get(
+                            matchViewModel.computerPlayer!!.computerPlayerId
+                        )!!
+                    )
+                },
+                scoreBasedUpdatesCallback = {
+                    matchViewModel.scoreBasedUpdates()
+                },
+
+                resultNavigationCallback = {
+                    resultNavigation()
+                },
+                resetUIScoresCallback = {
+                    resetScores()
+                }
+            )
+        }
     }
 
-    private fun gameOverNavigation() {
+
+    private fun resultNavigation() {
         if (matchViewModel.gameService?.isGameOver!!) {
-            findNavController().navigate(R.id.action_gameResultFragment_to_nav_home)
-            val text = "Game Over"
-            val duration = Toast.LENGTH_SHORT
-            val toast = Toast.makeText(context, text, duration)
-            toast.show()
-        } else {
             findNavController().navigate(R.id.action_gameFragment_to_gameResultFragment)
+        } else {
+            findNavController().navigate(R.id.action_gameFragment_self)
         }
     }
 
-    private fun selectComputerChoice(hand: Hand) {
+    private fun opponentMoveUIUpdate(hand: Hand) {
```

Comment:
This function is not really modular, but it will do for now.

Reply:
What do you mean by modular?
Because all it does is takes the hand (already chosen) of an opponent and update the view so its move is displayed (it's more bright the same way the color changes when a user chooses a hand). It is in Game Fragment because it only affects the view. 
I can change it but I don't think I understand the problem now.

Reply:
The only issue is using `binding.<hand>RBOpponent.isChecked = true` for every different hand. Because there is logic in common, you would want to do something like this.
```kotlin
fun getViewBindingForHand(hand: Hand): ch.epfl.sweng.rps.databinding.FragmentGameBinding? {
    when (hand) {
        Hand.ROCK -> binding.rockRBOpponent
        /* Same for other hands */
        Hand.NONE -> null
    }
}

private fun opponentMoveUIUpdate(hand: Hand) {
    /* Optionnally handle NONE in a different way here. */
    val binding = getViewBindingForHand(hand)
    binding?.isChecked = true
} 
```

Reply:
I fixed that now